util/rbac/rbac.go

```diff
@@ -188,10 +190,33 @@ func (e *Enforcer) EnableEnforce(s bool) {
 
 // LoadPolicy executes casbin.Enforcer functionality and will invalidate cache if required.
 func (e *Enforcer) LoadPolicy() error {
-	_, err := e.tryGetCabinEnforcer("", "")
+	_, err := e.tryGetCasbinEnforcer("", "")
 	return err
 }
 
+// CheckUserDefinedRoleReferentialIntegrity iterates over roles and policies to validate the existence of a matching policy subject for every defined role
+func CheckUserDefinedRoleReferentialIntegrity(e CasbinEnforcer) error {
+	allRoles, err := e.GetAllRoles()
+	if err != nil {
+		return err
+	}
+	notFound := make([]string, 0)
+	for _, roleName := range allRoles {
+		permissions, err := e.GetImplicitPermissionsForUser(roleName)
+		if err != nil {
+			return err
+		}
+		if len(permissions) > 0 {
```

Comment:
NIT: i think the code reads better like this, what do you think?

```suggestion
		if len(permissions) == 0 {
		    notFound = append(notFound, roleName)
		}
```

Reply:
Agree it is cleaner- implemented